feat: add routing and ability to navigate to a board

benox3 opened the pull request:
Resolves #37 

#### Tasks at a high level:
- add hash based routing
- link to a board page
- fetch board data
- routing decorator for storybook stories
- remove App placeholder story

benox3 commented on src/components/Boards/components/BoardTile/BoardTile.styles.tsx:
```diff
@@ -0,0 +1,25 @@
+import { styled } from '@material-ui/core/styles';
```
We can pull this out into shared styled components in the future.

remarkablemark replied:
What's the purpose of the muted link? Is this a stylistic choice? Let's make sure the app is still accessible

benox3 replied:
This just removes text decoration, shouldn't affect accessibility.

remarkablemark replied:
👍

remarkablemark commented on src/components/ErrorPage/ErrorPage.tsx:
```diff
@@ -0,0 +1,17 @@
+  return (
+    <Typography component="div">
+      <Box display="flex" height="100%" width="300" fontSize="h6.fontSize">
+        Oops... The link you are trying to reach can not be found
```
`s/can not/cannot/`

remarkablemark commented on src/components/ErrorPage/ErrorPage.tsx:
```diff
@@ -0,0 +1,17 @@
+import Box from '@material-ui/core/Box';
+import Typography from '@material-ui/core/Typography';
+
+export default function ErrorPage() {
```
Is error page solely for 404 error or does it encapsulate all kinds of errors?

benox3 replied:
Right now it's kind of a catch all, we can tweak once we have better handling in the future

remarkablemark replied:
Sounds good

remarkablemark commented:
LGTM. It seems that CI failed the build because of coveralls but I think we can create a follow-up task to enforce 100% coverage.

benox3 commented:
> LGTM. It seems that CI failed the build because of coveralls but I think we can create a follow-up task to enforce 100% coverage.

Added and fixed up some tests. I have a follow up PR for adding storyshots which should add coverage for all the components render functions as well
